arch/risc-v/src/common/riscv_cpuidlestack.c

```diff
@@ -50,64 +50,13 @@
  *   2. RISC-V requires a 16-byte stack alignment.
  */
 
-#if CONFIG_SMP_NCPUS > 1
-static uint8_t aligned_data(16) cpu1_idlestack[CONFIG_IDLETHREAD_STACKSIZE];
-#endif
-
-#if CONFIG_SMP_NCPUS > 2
-static uint8_t aligned_data(16) cpu2_idlestack[CONFIG_IDLETHREAD_STACKSIZE];
-#endif
-
-#if CONFIG_SMP_NCPUS > 3
-static uint8_t aligned_data(16) cpu3_idlestack[CONFIG_IDLETHREAD_STACKSIZE];
-#endif
-
-#if CONFIG_SMP_NCPUS > 4
-static uint8_t aligned_data(16) cpu4_idlestack[CONFIG_IDLETHREAD_STACKSIZE];
-#endif
-
-#if CONFIG_SMP_NCPUS > 5
-static uint8_t aligned_data(16) cpu5_idlestack[CONFIG_IDLETHREAD_STACKSIZE];
-#endif
-
-#if CONFIG_SMP_NCPUS > 6
-static uint8_t aligned_data(16) cpu6_idlestack[CONFIG_IDLETHREAD_STACKSIZE];
-#endif
-
-#if CONFIG_SMP_NCPUS > 7
-static uint8_t aligned_data(16) cpu7_idlestack[CONFIG_IDLETHREAD_STACKSIZE];
-#endif
+uint8_t aligned_data(16)
+g_cpux_idlestack[SMP_STACK_SIZE * (CONFIG_SMP_NCPUS - 1)];
```

Comment:
Maybe better to use
```suggestion
g_cpux_idlestack[SMP_STACK_SIZE][CONFIG_SMP_NCPUS - 1];
```
? And later change `stack_alloc = (uintptr_t)&g_cpux_idlestack[SMP_STACK_SIZE * (cpu - 1)];` to `stack_alloc = (uintptr_t)&g_cpux_idlestack[SMP_STACK_SIZE][cpu - 1];`, just for readability perspective

Reply:
Done